### DISCUSSION TITLE
Bullet Contact Stability for Robotics, Locomotion and RL

### DISCUSSION THREAD

**Discussion by Danfoa** [bulletphysics/bullet3#3946] [Q&A]
I have found very little information on the outdated documentation and the forums about the best contact parameters for locomotion and reinforcement learning. 

Sadly taking the default values from the Bullet Gym locomotion environments results in highly unstable contact dynamics even with very low time-step and a high number of solver iterations. I have found that the `contactERP` is highly influential in this, yet I found ZERO information on what this parameter means, and what is its range. 

Shall we use this thread to solve some of these questions and state the best practices for future users? 

For instance, 

- What parameters should change for a legged and wheeled robot? 
- What is the range, of contactERP, what does it meed, and how does it relates to the solverIterations and timeStep (Paper link?)

As for what I can test low contactERP are undesirable for locomotion as contacts like this occur:
![image](https://user-images.githubusercontent.com/8356912/130817695-0610e300-2d79-4e0b-b9e8-3a62c1286854.png)
This is with `contactEPR=0.08` `numSolverIterations=50` 200Hz 
_________________________
UPDATE 

1.  Information on the Contact ERP can be found in the [ODE Manua](http://ode.org/wiki/index.php?title=Manual#Joints_and_constraints). Its range is recommended between 0 and 0.8. Yet the recommended use depending on application is still an open question.

**Comment by tlemarec** [comment]
Sadly you wont find much help here, the documentation is really fragmented and with no real way to participate in.
And the pybullet maintainer are understaffed.
